File: backend/API/ollama-api-test.py
import requests
import json

url = "http://localhost:11434/api/generate"
payload = {
    "model": "llama3.2",
    "prompt": "Why is the sky blue?"
}

response = requests.post(url, json=payload)
if response.status_code == 200:
    # Handle streaming response
    for line in response.text.strip().split('\n'):
        try:
            data = json.loads(line)
            if "response" in data:
                print(data["response"], end="", flush=True)
        except json.JSONDecodeError:
            continue
else:
    print(f"Request failed with status code: {response.status_code}")

File: backend/API/ollama-api-test.js
const fetch = require('node-fetch');

const url = 'http://localhost:11434/api/generate';
const payload = {
    model: 'llama3.2',
    prompt: 'Why is the sky blue?'
};

async function generateResponse() {
    try {
        const response = await fetch(url, {
            method: 'POST',
            headers: {
                'Content-Type': 'application/json',
            },
            body: JSON.stringify(payload)
        });

        if (response.ok) {
            // Handle streaming response
            const reader = response.body;
            for await (const chunk of reader) {
                const lines = chunk.toString().split('\n');
                
                for (const line of lines) {
                    if (line.trim()) {
                        try {
                            const data = JSON.parse(line);
                            if (data.response) {
                                process.stdout.write(data.response);
                            }
                        } catch (error) {
                            continue;
                        }
                    }
                }
            }
        } else {
            console.log(`Request failed with status code: ${response.status}`);
        }
    } catch (error) {
        console.error('Error:', error);
    }
}

generateResponse();
